### PR TITLE
Update dependencies to fix security warnings

### LINKED ISSUE #224: Replace optimist
Replace optimist with a better component. As it's no longer support by the developer and doesn't met the requirements anymore.
## 

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8544676-replace-optimist?utm_campaign=plugin&utm_content=tracker%2F73887&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F73887&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by rfrm**
After running `npm audit fix` there are still 8 vulnerabilities that have to be reviewed manually. The issue is caused by the `optimist` and `lab` libraries dependencies. The fix was to:

1. Replace optimist with yargs.
2. Update lab from v14.3.4 to v18.1.2

This closes #224

### CODE REVIEW

**Comment by rfrm** [comment]
Fixes https://github.com/pagerinc/api-scheduling/issues/341

**Comment by aorinevo** [comment]
This PR would also close https://github.com/db-migrate/node-db-migrate/issues/674

**Comment by wzrdtales** [comment]
It is passing the tests, that is a good sign, the other PR made them fail. Since you made some changes to the tests itself due to the update of lab, I will review those first.

File: test/integration/create_test.js [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+const Lab = require('@hapi/lab');
 const lab = (exports.lab = Lab.script());
 const fs = require('fs');
 const path = require('path');
 const cp = require('child_process');
+const util = require('util');
 const dbmUtil = require('db-migrate-shared').util;
 
-const rmdir = require('rimraf');
+const rmdir = util.promisify(require('rimraf'));
```
Comment (wzrdtales) [inline test/integration/create_test.js:12]: since we use project wide bluebird, please use bluebird promisify function.

Comment (aorinevo) [reply test/integration/create_test.js]: I'd advocate for ease of use and less dependencies over, IMO, marginal performance gains - this is specific to the purpose this package serves.  

I'd also note that performance is a function of the node version.  The delta in performance certainly increases as we fallback towards older and older versions of node.

Thoughts on when we can move to use native promisify and promises?

**Comment by BorntraegerMarc** [comment]
@wzrdtales should a new version be released so we can update & fix our security issues?

**Comment by wzrdtales** [comment]
It needs a backport to `v0.11.x` first. I am thinking if I can go ahead maybe with the current `master` tree already and am looking into what has to be done for that.

**Comment by wzrdtales** [comment]
@BorntraegerMarc especially looking which breaking changes that means, if it is not feasible I will backport.

**Comment by wzrdtales** [comment]
decided to backport now, I don't want to rush anything here. there is a new 0.11.7 out right now, let me know should there be any troubles.

**Comment by wzrdtales** [comment]
to be exact, `v1.0.0` or the master branch as of now is deprecating older node versions and drops support for them entirely. The backport shouldn't change anything about the support level of supported versions for 0.11.x. As of now on top of my head I don't saw another breaking change other than modernizing the code base to a newer standard. everything else (as always) I managed to broadly keep backwards compatible. However, I will go sure to check this one more time.

**Comment by BorntraegerMarc** [comment]
@wzrdtales awesome 🎉 I updated this lib in our project & our CI passes. LGTM 🙂

**Comment by BorntraegerMarc** [comment]
hehe looks like CI doesn't catch all the problems @wzrdtales you forgot to remove a debug statement in this commit: https://github.com/db-migrate/node-db-migrate/commit/8b5beacdee33ce0a63805ec453de959ef689bf90#diff-e4e7fada050c1319260e0dda7821fc5aR10

happens to the best of us 😉

**Comment by wzrdtales** [comment]
yah, removed that one on the master already, didn't realized it slipped through into v0.11.x too, i will fix that now

**Comment by wzrdtales** [comment]
I will put `rg console.log` into my workflow for releasing^^ https://github.com/db-migrate/node-db-migrate/commit/ce77a285a3fe4dc728e81f7e140b5dc4e84a8a40#diff-b4e530c41bbf801b4ef6f4840a3b01aaL127 that one also slipped through

**Comment by wzrdtales** [comment]
`0.11.9` is out though

**Pull request merged**